Sources/TokamakCore/CGStubs.swift

```diff
@@ -16,3 +16,194 @@
 //
 
 public typealias CGFloat = Double
+public struct CGPoint: Equatable {
+  public let x: CGFloat
+  public let y: CGFloat
+
+  public init(x: CGFloat, y: CGFloat) {
+    self.x = x
+    self.y = y
+  }
+
+  public static var zero: Self {
+    .init(x: 0, y: 0)
+  }
+}
+
+public struct CGSize: Equatable {
+  public let width: CGFloat
+  public let height: CGFloat
+
+  public init(width: CGFloat, height: CGFloat) {
+    self.width = width
+    self.height = height
+  }
+
+  public static var zero: Self {
+    .init(width: 0, height: 0)
+  }
+}
+
+public struct CGRect: Equatable {
+  public let origin: CGPoint
+  public let size: CGSize
+
+  public init(_ origin: CGPoint, _ size: CGSize) {
+    self.origin = origin
+    self.size = size
+  }
+
+  public static var zero: Self {
+    .init(.zero, .zero)
+  }
+}
+
+public enum CGLineCap {
+  /// A line with a squared-off end. Extends to the endpoint of the Path.
+  case butt
+  /// A line with a rounded end. Extends past the endpoint of the Path.
+  case round
+  /// A line with a squared-off end. Extends past the endpoint of the Path.
+  case square
+}
+
+public enum CGLineJoin {
+  case miter
+  /// A join with a rounded end. Extends past the endpoint of the Path.
+  case round
+  /// A join with a squared-off end. Extends past the endpoint of the Path.
+  case bevel
+}
+
+/// An affine transformation matrix for use in drawing 2D graphics.
+///
+///     a   b   0
+///     c   d   0
+///     tx  ty  1
+public struct CGAffineTransform: Equatable {
+  public var a: CGFloat
+  public var b: CGFloat
+  public var c: CGFloat
+  public var d: CGFloat
+  public var tx: CGFloat
+  public var ty: CGFloat
+
+  /// The identity matrix
+  public static let identity: Self = .init(a: 1, b: 0, // 0
+                                           c: 0, d: 1, // 0
+                                           tx: 0, ty: 0) // 1
+
+  public init(a: CGFloat, b: CGFloat,
+              c: CGFloat, d: CGFloat,
+              tx: CGFloat, ty: CGFloat) {
+    self.a = a
+    self.b = b
+    self.c = c
+    self.d = d
+    self.tx = tx
+    self.ty = ty
+  }
+
+  /// Returns an affine transformation matrix constructed from a rotation value you provide.
+  /// - Parameters:
+  ///   - angle: The angle, in radians, by which this matrix rotates the coordinate system axes.
+  ///            A positive value specifies clockwise rotation and a negative value specifies counterclockwise rotation.
+  public init(rotationAngle angle: CGFloat) {
+    // FIXME: How do we use sin/cos in Wasm?
+    //        With Darwin: self.init(a: cos(angle), b: sin(angle), c: -sin(angle), d: cos(angle), tx: 0, ty: 0)
```

Comment:
> WebAssembly doesn’t include its own math functions like sin, cos, exp, pow, and so on. WebAssembly’s strategy for such functions is to allow them to be implemented as library routines in WebAssembly itself (note that x86’s sin and cos instructions are slow and imprecise and are generally avoided these days anyway). Users wishing to use faster and less precise math functions on WebAssembly can simply select a math library implementation which does so.

([WASM FAQ](https://webassembly.org/docs/faq/))

Reply:
Maybe best off hand-coding it? See [Wikipedia](https://en.wikipedia.org/wiki/Lookup_table#Computing_sines) for one way based on a Taylor series.

Reply:
These are available in [WASI](wasi.dev), which you currently can get with `import Glibc`. I do plan to rename that module to `WASI`, so you will have to write `import WASI` in the future, but for now `import Glibc` should work.